packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart

```diff
@@ -238,6 +239,8 @@ class SignInStateMachine extends AuthStateMachine<SignInEvent, SignInState> {
         ..poolId = config.poolId
         ..deviceKey = user.deviceSecrets?.deviceKey
         ..challengeParameters = challengeParameters
+        // Passing in the Builder causes issues with serialization
+        ..analyticsMetadata = get<AnalyticsMetadataType>()
```

Comment:
Instead of passing it to the workers (since they don't really need it) and to each response individually, you could add it once here: https://github.com/aws-amplify/amplify-flutter/blob/cc8f3bdb049142a8171b8696d240e212196077fa/packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart#L819